src/internal/components/LiveRegion.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import {VisuallyHidden} from './VisuallyHidden'
+
+type LiveRegionContext = {
+  announce: (message: string) => void
+  message: string
+}
+
+const LiveRegionContext = React.createContext<LiveRegionContext | null>(null)
+
+function useLiveRegion() {
+  const context = React.useContext(LiveRegionContext)
+  if (!context) {
+    throw new Error('useLiveRegion() must be used within a <LiveRegion>')
+  }
+  return context
+}
+
+function LiveRegion({children}: React.PropsWithChildren) {
+  const [message, setMessage] = React.useState('')
+  const value = React.useMemo(() => {
+    return {
+      announce: setMessage,
+      message,
+    }
+  }, [message])
+
+  return <LiveRegionContext.Provider value={value}>{children}</LiveRegionContext.Provider>
+}
+
+function LiveRegionOutlet() {
+  const liveRegion = useLiveRegion()
+  return (
+    <VisuallyHidden role="status" aria-live="polite" aria-atomic={true}>
+      {liveRegion.message}
+    </VisuallyHidden>
+  )
+}
+
+function Message({value}: {value: string}) {
```

Comment:
Is this a declarative API for making announcements?

Reply:
@colebemis exactly, still trying to see if it works the way we anticipate 😅 Thankfully everything announces as-intended right now but a little worried about the declarative API translating to side-effects with announcements.

Reply:
I love it!!